config/config.go

```diff
@@ -1,24 +1,21 @@
 package config
 
 import (
-	"bytes"
 	_ "embed"
-	"encoding/json"
 	"errors"
 	"fmt"
-	"io"
 	"reflect"
 	"regexp"
 	"strings"
 
-	"github.com/ghodss/yaml"
 	"github.com/invopop/jsonschema"
+	"sigs.k8s.io/yaml"
```

Comment:
Why do we use that instead of ghodss/yaml? This will cause more dependencies on the loft-sh/vcluster-config side, is that intended?

Reply:
The current version of `ghodss/yaml` we depend on does not provide a strict marshaller that errors on unknown fields. But you are right, it doesn't make sense to import both `ghodss/yaml` and `sigs.k8s.io/yaml`. I could replace `ghodss/yaml` with the k8s yaml in following files, so that we only have one yaml dep? WDYT?

- https://github.com/loft-sh/vcluster/blob/main/config/diff.go#L9
- https://github.com/loft-sh/vcluster/blob/main/config/legacyconfig/migrate.go#L9